core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,331 @@
+      throw new Error(`Card name ${cardName} not found in the graph.`);
+    }
+
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileStateGraphResizeButton);
+      await this.page.waitForSelector(mobileStateNodeSelector);
+      const elements = await this.page.$$(mobileStateNodeSelector);
+      await elements[cardIndex + elements.length / 2].click();
+    } else {
+      await this.page.waitForSelector(stateNodeSelector);
+      const elements = await this.page.$$(stateNodeSelector);
+      await elements[cardIndex].click();
+    }
+
+    await this.page.waitForNetworkIdle({idleTime: 700});
+  }
+
+  /**
+   * Function to add responses to the interactions. Currently, it only handles 'Number Input' interaction type.
+   * @param {string} interactionType - The type of the interaction.
+   * @param {string} answer - The response to be added.
+   * @param {string} feedback - The feedback for the response.
+   * @param {string} destination - The destination state for the response.
+   * @param {string} correctness - Whether the response is marked as correct.
+   */
+  async addResponsesToTheInteraction(
+    interactionType: string,
+    answer: string,
+    feedback: string,
+    destination: string,
+    correctness: string
+  ): Promise<void> {
+    switch (interactionType) {
+      case 'Number Input':
+        await this.type(floatFormInput, answer);
+        break;
+      // Add cases for other interaction types here
+      // case 'otherInteractionType':
+      //   await this.type(otherFormInput, answer);
+      //   break;
+      default:
+        throw new Error(`Unsupported interaction type: ${interactionType}`);
+    }
+
+    await this.clickOn(feedbackEditorSelector);
+    await this.type(stateContentInputField, feedback);
+    // The '/' value is used to select the 'a new card called' option in the dropdown.
+    await this.select(destinationCardSelector, '/');
+    await this.type(addStateInput, destination);
+    if (correctness === 'correct') {
+      await this.clickOn(correctAnswerInTheGroupSelector);
+    }
+    await this.clickOn(addNewResponseButton);
+    await this.page.waitForSelector(responseModalHeaderSelector, {
+      hidden: true,
+    });
+  }
+
+  /**
+   * Function to navigate to the preview tab.
+   */
+  async navigateToPreviewTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileNavbarDropdown);
+      await this.page.waitForSelector(mobileNavbarPane);
+      await this.clickOn(mobilePreviewTabButton);
+    } else {
+      await this.clickOn(previewTabButton);
+    }
+    await this.page.waitForNavigation();
+  }
+
+  /**
+   * Function to verify if the preview is on a particular card by checking the content of the card.
+   * @param {string} cardName - The name of the card to check.
+   * @param {string} text - The expected text content of the card.
+   */
+  async expectPreviewCardContentToBe(
+    cardName: string,
+    text: string
```

Comment:
expectedCardContent perhaps (put "expected" before values that communicate the expected behaviour, and match the names -- don't call one thing "content" and the other "text" if they both refer to the same thing).

Reply:
Done